server/channels/store/sqlstore/user_store.go

```diff
@@ -1674,7 +1674,27 @@ func (us SqlUserStore) performSearch(query sq.SelectBuilder, term string, option
 
 func (us SqlUserStore) AnalyticsGetInactiveUsersCount() (int64, error) {
 	var count int64
-	err := us.GetReplicaX().Get(&count, "SELECT COUNT(Id) FROM Users WHERE DeleteAt > 0")
+	query := us.getQueryBuilder().
+		Select("COUNT(Id)").
+		From("Users")
+	if us.DriverName() == model.DatabaseDriverPostgres {
+		query = query.LeftJoin("Bots ON Users.ID = Bots.UserId").
+			Where(sq.And{
+				sq.Gt{"Users.DeleteAt": 0},
+				sq.Eq{"Bots.UserId": nil},
+			})
+	} else {
+		query = query.Where(sq.And{
+			sq.Expr("Users.Id NOT IN (SELECT UserId FROM Bots)"),
+			sq.Gt{"Users.DeleteAt": 0},
+		})
+	}
+	queryStr, args, err := query.ToSql()
+	if err != nil {
+		return int64(0), errors.Wrap(err, "failed to create a SQL query to count inactive users")
+	}
+	err = us.GetReplicaX().Get(&count, queryStr, args...)
```

Comment:
1/5 that the query separation is not worth it. MySQL is deprecated anyway, so the code won't be here for long. It's also a query that is not run very often, so I don't think the performance gains are worth it.

Reply:
it is deprecated but we keep supporting it until v11, and on top of that there will be ESRs being supported for a while, so it is a longer period than a couple of months.

Now, for the performance gains, I'm not entirely sure how much gain do we get from this. The original effort was https://mattermost.atlassian.net/browse/MM-53438 , and it looks like a big gain, but again, this is an endpoint less used that the one on user's count

Reply:
If we care about performance, we should  call `SqlUserStore.Count` as that method's performance has already been optimised.

Reply:
Good point, I missed that `Count` doesn't support counting _only_ the inactive users.

Reply:
I love to have this kind of conversations, forces me to explore further. Thanks!